src/components/RoomNameInput.js

```diff
@@ -131,9 +139,16 @@ class RoomNameInput extends Component {
                 prefixCharacter="#"
                 placeholder={this.props.translate('newRoomPage.social')}
                 containerStyles={[styles.mb5]}
-                onChangeText={roomName => this.checkAndModifyRoomName(roomName)}
+                onChangeText={(roomName) => {
+                    const modifiedRoomName = this.modifyRoomName(roomName);
+                    this.setState({roomName: modifiedRoomName});
+                    if (!this.props.shouldShowErrorOnChange) {
+                        return;
+                    }
+                    this.validateRoomName(modifiedRoomName);
+                }}
                 value={this.state.roomName.substring(1)}
-                errorText={this.state.error}
+                errorText={this.props.shouldShowErrorOnChange ? this.state.error : this.props.errorText}
```

Comment:
Error text should come from the parent not be set to state. Inputs should not handle their own validation.